source/mge.js

```diff
@@ -0,0 +1,988 @@
+
+    set metadata(value) {
+        this._metadata = value;
+    }
+
+    set exception(value) {
+        this._exceptionCallback = value;
+    }
+
+    get format() {
+        this._unpickle();
+        var version = "1.6";
+        var dump_info = this._graphs[0].data.dump_info;
+        if (dump_info !== undefined) {
+            version = dump_info.version;
+        }
+        return 'megengine v' + version;
+    }
+
+    get graphs() {
+        this._unpickle();
+        return this._graphs;
+    }
+
+    get littleEndian() {
+        this._unpickle();
+        return this._littleEndian;
+    }
+
+    _unpickle() {
+        if (!this._stream) {
+            return;
+        }
+
+        const data = this._stream.length < 0x7ffff000 ? this._stream.peek() : this._stream;
+        const execution = new megengine.Execution(null, this._exceptionCallback);
+        const unpickler = python.Unpickler.open(data, execution);
+
+        this._stream = null;
+        this._exceptionCallback = null;
+
+        const module_source_map = new Map();
+        const deserialized_objects = new Map();
+        unpickler.persistent_load = (saved_id) => {
+            const typename = saved_id.shift();
+            const data = saved_id;
+            switch (typename) {
+                case 'module': {
+                    const module = data[0];
+                    const source = data[2];
+                    module_source_map.set(module, source);
+                    return data[0];
+                }
+                case 'storage': {
+                    const name = data.shift();
+                    const storage_type = execution.type(name);
+                    const root_key = data.shift();
+                    data.shift(); // location
+                    const size = data.shift();
+                    const view_metadata = data.shift();
+                    if (!deserialized_objects.has(root_key)) {
+                        const obj = new storage_type(size);
+                        deserialized_objects.set(root_key, obj);
+                    }
+                    if (view_metadata) {
+                        const view_key = view_metadata.shift();
+                        view_metadata.shift(); // view_offset
+                        view_metadata.shift(); // view_size
+                        if (!deserialized_objects.has(view_key)) {
+                            const view = null; // storage.slice(view_offset, view_offset + view_size);
+                            deserialized_objects.set(view_key, view);
+                        }
+                        return deserialized_objects.get(view_key);
+                    }
+                    return deserialized_objects.get(root_key);
+                }
+                default: {
+                    throw new megengine.Error("Unsupported persistent load type '" + typename + "'.");
+                }
+            }
+        };
+        const obj = unpickler.load();
+        if (!obj) {
```

Comment:
Inline `find` and `findModule` method, simplify.

Reply:
done